google/api_core/gapic_v1/method.py

```diff
@@ -18,14 +18,26 @@
 pagination, and long-running operations to gRPC methods.
 """
 
+import enum
 import functools
 
 from google.api_core import grpc_helpers
 from google.api_core import timeout
 from google.api_core.gapic_v1 import client_info
 
 USE_DEFAULT_METADATA = object()
-DEFAULT = object()
+
+
+class _MethodDefault(enum.Enum):
```

Comment:
Since this needs to be used in the generated code, should we make this public?

Reply:
I figured that the generator is in the "friends zone" for api-core. :)

Reply:
I think the concern is that this type will be exposed as a type to users of the API?

Reply:
It seems like an "implementation detail" to me, uninteresting to anyone.  It will be in generated code, and present only in the definition of the `OptionalRetry` type -- see https://github.com/googleapis/gapic-generator-python/pull/1032#pullrequestreview-780331392.  I am in fact considering a change to that PR which removes even the classname.  Instead of:

```python
try:
    OptionalRetry = Union[retries.Retry, gapic_v1.method._MethodDefault]
except AttributeError:
    OptionalRetry = Union[retries.Retry, object]
```

I'm thinking this would be simpler / clearer:
```python
OptionalRetry = Union[retries.Retry, type(gapic_v1.method.DEFAULT)]
```

Reply:
> I think the concern is that this type will be exposed as a type to users of the API?

Yes, that is a concern.

On second thought though, I think most users usages of the `_MethodDefault` type will be by omitting the `retry` argument, so they probably won't need to access this directly. Especially not if `OptionalRetry` is exposed in the generated client.